_search-plugins/searching-data/retrieve-specific-fields.md

```diff
@@ -0,0 +1,908 @@
+
+1. Index your data:
+
+    ```json
+    PUT my_index/_doc/1
+    {
+      "title": "OpenSearch Basics",
+      "author": "John Doe",
+      "publication_date": "2021-01-01",
+      "price": 29.99
+    }
+    ```
+    {% include copy-curl.html %}
+
+2. Perform a search using source filtering:
+
+    ```json
+    POST my_index/_search
+    {
+      "_source": ["title", "author"],
+      "query": {
+        "match_all": {}
+      }
+    }
+    ```
+    {% include copy-curl.html %}
+
+Expected response:
+
+```json
+{
+  "hits": {
+    "total": {
+      "value": 1,
+      "relation": "eq"
+    },
+    "max_score": 1.0,
+    "hits": [
+      {
+        "_index": "my_index",
+        "_id": "1",
+        "_score": 1.0,
+        "_source": {
+          "title": "OpenSearch Basics",
+          "author": "John Doe"
+        }
+      }
+    ]
+  }
+}
+```
+
+### Excluding fields with source filtering
+
+You can choose to exclude fields using `"excludes"` parameter in search request, see following example:
+
+```json
+POST my_index/_search
+{
+  "_source": {
+    "excludes": ["price"]
+  },
+  "query": {
+    "match_all": {}
+  }
+}
+```
+{% include copy-curl.html %}
+
+Expected response:
+
+```json
+{
+  "hits": {
+    "total": {
+      "value": 1,
+      "relation": "eq"
+    },
+    "max_score": 1.0,
+    "hits": [
+      {
+        "_index": "my_index",
+        "_id": "1",
+        "_score": 1.0,
+        "_source": {
+          "title": "OpenSearch Basics",
+          "author": "John Doe",
+          "publication_date": "2021-01-01"
+        }
+      }
+    ]
+  }
+}
+```
+
+### Including and excluding fields in the same search
+
+There might be cases where both `include` and `exclude` parameters are necessary. The following examples demonstrate the usefulness of this option.
+
+Consider a`products` index containing the following document:
+
+```json
+{
+  "product_id": "123",
+  "name": "Smartphone",
+  "category": "Electronics",
+  "price": 699.99,
+  "description": "A powerful smartphone with a sleek design.",
+  "reviews": [
+    {
+      "user": "john_doe",
+      "rating": 5,
+      "comment": "Great phone!",
+      "date": "2023-01-01"
+    },
+    {
+      "user": "jane_doe",
+      "rating": 4,
+      "comment": "Good value for money.",
+      "date": "2023-02-15"
+    }
+  ],
+  "supplier": {
+    "name": "TechCorp",
+    "contact_email": "support@techcorp.com",
+    "address": {
+      "street": "123 Tech St",
+      "city": "Techville",
+      "zipcode": "12345"
+    }
+  },
+  "inventory": {
+    "stock": 50,
+    "warehouse_location": "A1"
+  }
+}
+```
+
+You want to perform a search on this index but include only the `name`, `price`, `reviews`, and `supplier` fields in the response. Additionally, you want to exclude any `contact_email` fields from the `supplier` object and `comment` fields from the `reviews` object. To achieve this, run the following search:
```

Comment:
Nit: "You want to" (two instances) isn't really our usual voice, but I'll allow it if this is standard syntax or truly the best way to express this.